common/firmware/i2c/i2c.c

```diff
@@ -0,0 +1,67 @@
+#include "common/firmware/i2c.h"
+
+static void Error_Handler(void);
+
+void HAL_I2C_MspInit(I2C_HandleTypeDef* hi2c1) {
+    __HAL_RCC_I2C1_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+
+    // PIN PB7 is SCL
+    // PIN PB8-BOOT0 is SDA
+    if(hi2c1->Instance==I2C1) {
+        GPIO_InitTypeDef GPIO_InitStruct = {0};
+        GPIO_InitStruct.Pin = GPIO_PIN_7 | GPIO_PIN_8;
+        GPIO_InitStruct.Mode = GPIO_MODE_AF_OD;
+        GPIO_InitStruct.Pull = GPIO_PULLUP;
+        GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+        GPIO_InitStruct.Alternate = GPIO_AF4_I2C1;
+        HAL_GPIO_Init(
+            GPIOB,  // NOLINT(cppcoreguidelines-pro-type-cstyle-cast)
+            &GPIO_InitStruct);  // NOLINT(cppcoreguidelines-pro-type-cstyle-cast)
+    }
+}
+
+I2C_HandleTypeDef MX_I2C1_Init()
+{
+    I2C_HandleTypeDef hi2c1;
```

Comment:
Just as we discussed in some other PRs, if this is a common library it really can't do things like directly configure a specific I2C peripheral instance. Using this code would require every program to use one specific I2C peripheralk, and to map its outputs to the same specific GPIO pins. It should either be part of `pipette`, because I think that's where it's used, or have a lot of the extreme low level configuration pulled out - at least the I2C unit assignment and the specific GPIO pins.